File: corecomply-mvp/web/src/types/integration.ts
export interface Integration {
  id: number;
  name: string;
  type: string;
  status: string;
  lastSync?: string | null;
  configuration?: string | null;
  apiKey?: string | null;
  endpoint?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CreateIntegrationRequest {
  name: string;
  type: string;
  configuration?: string | null;
  apiKey?: string | null;
  endpoint?: string | null;
}

export interface UpdateIntegrationRequest {
  name: string;
  type: string;
  status: string;
  configuration?: string | null;
  apiKey?: string | null;
  endpoint?: string | null;
}

export interface SyncResponse {
  message: string;
  lastSync: string;
}

File: corecomply-mvp/api/CoreComply.Api/Domain/Entities/Integration.cs
namespace CoreComply.Api.Domain.Entities;

public class Integration
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty; // payroll, hr, accounting, api
    public string Status { get; set; } = "disconnected"; // connected, disconnected, error, configuring
    public DateTime? LastSync { get; set; }
    public string? Configuration { get; set; } // JSON configuration
    public string? ApiKey { get; set; }
    public string? Endpoint { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
